### PR TITLE
Changes on frontend related to proper working with backend side

### PR DESCRIPTION
**Pull request by Mats30**
Added:
* BooksAPI interface to get request result

Modified:
* Book-list component adapted to connecting with back-end side
* Book entity adpated to database structure
* New method used to building request with params (partly cover sorting feature)
* Working paging and partly sorting

Removed:
* Utility service
* Whitespaces and unused imports

### CODE REVIEW

File: robot/src/app/components/books-list/books-list.component.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
+.container {
```
Comment (szczepanskikrs) [inline robot/src/app/components/books-list/books-list.component.css]: :D WOW i understand this part !

File: robot/src/app/services/book.service.ts [hunk 2]
```diff
@@ -1,62 +1,26 @@
...
-import {Book} from "../entities/book";
+import {HttpClient} from '@angular/common/http';
+import {Observable} from "rxjs/Observable";
+import {BooksApi} from "../entities/booksApi";
+import {isUndefined} from "util";
 
 @Injectable()
 export class BookService {
-  data: Array<Book>;
+  private host: string = "http://localhost:8080/database-1.0-SNAPSHOT/books";
```
Comment (szczepanskikrs) [inline robot/src/app/services/book.service.ts]: 8088 port on vps but I can change it back to 8080 if you want :+1:

File: robot/src/app/services/book.service.ts [hunk 3]
```diff
@@ -1,62 +1,26 @@
...
-    let potop: Book = {
-      title: "Potop",
-      author: "Sienkiewicz",
-      bookshop: "Bonito",
-      category: "powieść",
-      new_price: 49.99,
-      old_price: 69.99,
-      url : "https://bonito.pl/k-1316948-potop"
-    };
+  public getPagedBooksList(page: number, numberInPage: number, sort: string): Observable<BooksApi> {
```
Comment (szczepanskikrs) [inline robot/src/app/services/book.service.ts]: I like typescript looks alot like java

File: robot/src/app/services/book.service.ts (hunk 2 above)
Comment (Mats30) [reply robot/src/app/services/book.service.ts]: Sorry, in my local tests I used 8080 port :)

**Pull request merged**